src/internal/components/InternalThemeEditor.tsx

```diff
@@ -62,35 +63,40 @@ export const InternalThemeEditor = ({
     }
 
     const newThemeValues = {
-      ...themeHistory.history[themeHistory.index],
       ...generateCssVariablesFromPuckFields(newValue, topLevelKey),
     };
 
     const newHistory = {
-      history: [...themeHistory.history, newThemeValues],
-      index: themeHistory.index + 1,
+      histories: [
+        ...themeHistory.histories,
+        { id: uuidv4(), state: { data: newThemeValues } },
+      ] as History[],
+      index: themeHistory.histories.length,
```

Comment:
Should this be length - 1?

Reply:
ah not this one cuz it should be themeHistory.histories.length -1 + 1 (for the new history)